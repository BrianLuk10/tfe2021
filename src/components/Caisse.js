import React from 'react';

export default function Caisse() {
  return(
    <h2>Caisse</h2>
  );
}