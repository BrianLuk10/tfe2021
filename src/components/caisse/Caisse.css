.image{
    max-height: 200px;
    max-width: 200px;
    min-height: 200px;
    min-width: 200px;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
}

.container {
    display: grid;
    grid-auto-columns: max-content;
    grid-auto-flow: dense; 
    grid-auto-rows: minmax(100px, auto);
    grid-gap: 20px;
    grid-template-columns: repeat(2, 1fr);
    
  }
  .grid-item {
    grid-column: span 1;
    grid-row: span 1;
  }


.pagination {
  margin: 10px auto;
  display: flex;
  list-style: none;
  outline: none;
}
.pagination > .active > a{
  background-color: #47ccde ;
  border-color: #47ccde ;
  color: #fff;
}
.pagination > li > a{
  border: 1px solid #47ccde ;
  padding: 5px 10px;
  outline: none;
  cursor: pointer;
}
.pagination > .active > a, .pagination > .active > span, .pagination > .active > a:hover, .pagination > .active > span:hover, .pagination > .active > a:focus, .pagination > .active > span:focus{
  background-color: #47ccde ;
  border-color: #47ccde;
  outline: none ;
}
.pagination > li > a, .pagination > li > span{
  color: #47ccde
}
.pagination > li:first-child > a, .pagination > li:first-child > span, .pagination > li:last-child > a, .pagination > li:last-child > span{
  border-radius: unset
}


.modal {
  position: fixed;
  top: 0;
  left: 0;
  width:100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
}

.modal-main {
  position:fixed;
  background: white;
  width: 200px;
  height: auto;
  top:45%;
  left:50%;
  transform: translate(-50%,-50%);
}

.display-block {
  display: block;
}

.display-none {
  display: none;
}