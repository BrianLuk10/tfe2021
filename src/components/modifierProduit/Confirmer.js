import React from "react";

/**
 * function pour afficher le message de confirmation car la fonction alert() ne fonctionne pas avec Electron.js => BUG
 * @param {*} show
 * @param {*} confirmer
 * @param {boolean} si show est false affiche rien sinon affiche le message de confirmation
 * @returns {html} message de confimation
 */
export default function Confirmer({ show, confirmer }) {
  if (!show) {
    return <></>;
  }
  return (
    <div className="overlay">
      <div className="dialog">
        <div className="dialog__content">
          <h2 className="dialog__title">Action effectu√©</h2>
        </div>
        <hr />
        <div className="dialog__footer">
          <button className="dialog__confirm" onClick={confirmer}>
            Confirmer
          </button>
        </div>
      </div>
    </div>
  );
}
