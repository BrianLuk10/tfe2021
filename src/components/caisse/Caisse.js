import React from "react";
import { BrowserRouter, Route, Switch, Link } from "react-router-dom";
import axios from "axios";
import ReactPaginate from "react-paginate";
import "./Caisse.css";
import ReactToPrint from "react-to-print";

//initialisation du prix total à 0 lorsqu'on arrive vers cette page
let prixTotal = 0;

//création du Modal des produit, renvoie un modal avec un choix de 1 à 10 du produit
const Modal = ({ handleClose, validerModal, show, children }) => {
  const showHideClassName = show ? "modal display-block" : "modal display-none";

  return (
    <div className={showHideClassName}>
      <section className="modal-main">
        {children}
        <input
          type="range"
          min="1"
          max="10"
          class="slider"
          id="myRange"
        ></input>
        <div id="result"></div>
        <span></span>
        <button className="btn btn-success" onClick={validerModal}>
          Valider
        </button>
        <button className="btn btn-danger" onClick={handleClose}>
          Fermer
        </button>
      </section>
    </div>
  );
};


//création du Modal si le stock du produit est à 0, renvoie 'plus disponible'
const Modal2 = ({ handleClose2, show2 }) => {
  const showHideClassName2 = show2
    ? "modal display-block"
    : "modal display-none";

  return (
    <div className={showHideClassName2}>
      <section className="modal-main">
        <div>plus disponible</div>
        <span></span>
        <button className="btn btn-danger" onClick={handleClose2}>
          Fermer
        </button>
      </section>
    </div>
  );
};

export default class Caisse extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      posts: [],
      offset: 0,
      data: [],
      perPage: 14,
      currentPage: 0,
      caisse: [],
      categorie: "",
      value: "Tout",
      show: false,
      show2: false,
      article: [],
    };
    this.onChangeValue = this.onChangeValue.bind(this);
    this.showModal = this.showModal.bind(this);
    this.hideModal = this.hideModal.bind(this);
    this.hideModal2 = this.hideModal2.bind(this);
    this.handlePageClick = this.handlePageClick.bind(this);
  }

  //fonciton qui per;et d'afficher le modal des produits, est appelé lorsqu'on clique sur l'image du produit dans la caisse
  showModal = (pd) => {
    if (pd.stock > 0) {
      var slider = document.getElementById("myRange");
      var output = document.getElementById("result");
      output.innerHTML = slider.value;
      slider.oninput = function () {
        output.innerHTML = this.value;
      };
      this.setState({ show: true });
      sessionStorage.setItem("caisseProduit", JSON.stringify(pd));
      var data = JSON.parse(sessionStorage.getItem("caisseProduit"));
      document.getElementById("nom").innerHTML = data.nom;
      document.getElementById("prix").innerHTML = data.prix.toFixed(2) + "$";
      document.getElementById("img1").innerHTML =
        `<img class="image" src=` + data.image + `>`;
    } else {
      this.setState({ show2: true });
    }
  };

  //fonction qui remet l'etat de la caisse à 0 et sauvegarde l'etat de la caisse actuelle dans un sessionStorage, est appelé une fois la caisse fini
  reset = () => {
    sessionStorage.setItem("save", JSON.stringify(this.state.caisse));
    axios.post("http://localhost:3030/commande");
    for (let i = 0; this.state.caisse.length > i; i++) {
      axios.put(
        "http://localhost:3030/fournissement/stock/" + this.state.caisse[i].id,
        {
          nombre: this.state.caisse[i].nombre,
        }
      );
    }
    for (let i = 0; this.state.caisse.length > i; i++) {
      axios.post("http://localhost:3030/commande2", {
        id_produits: this.state.caisse[i].id,
        nombre: this.state.caisse[i].nombre,
      });
    }

    this.setState({ caisse: [] });
    prixTotal = 0;
    this.componentDidMount();
  };

  //cache le modal des produit
  hideModal = () => {
    this.setState({ show: false });
  };

  //cache le modal 'plus disponible'
  hideModal2 = () => {
    this.setState({ show2: false });
  };

  //une fois le modal valider, le cache et ajoute les données dans la caisse avec un sessionStorage
  validerModal = () => {
    this.setState({ show: false });
    var slider = document.getElementById("myRange");
    let data = JSON.parse(sessionStorage.getItem("caisseProduit"));
    data.nombre = slider.value;
    if (data.nombre > data.stock) {
    } else {
      sessionStorage.setItem("caisse", JSON.stringify(data));
      this.state.caisse.push(JSON.parse(sessionStorage.getItem("caisse")));
      this.componentDidMount();
    }
  };

  //fonction simple de calcul pour calculer le prix total de la caisse
  calculerPrixTotal = (a, b) => {
    let prix = a * b;
    prixTotal += prix;
  };

  //fonction qui change la value du state lorsqu'on la change, est utilisé pour les SELECT/OPTION html pour les catégories des produits.
  onChangeValue(e) {
    this.setState({ value: e.target.value });
    this.componentDidMount();
  }

  //fonction componentDidMount, méthode de React pour recevoir les données des produits de la caisse via une API et les affichent.
  componentDidMount() {
    axios.get(`http://localhost:3030/produit`).then((res) => {
      let filter;
      if (this.state.value == "Tout") {
        filter = res.data;
      } else {
        filter = res.data.filter(
          (x) => x.categorie_produits == this.state.value
        );
      }
      const posts = filter.map((obj) => ({
        id: obj.id_produits,
        nom: obj.nom_produits,
        prix: obj.prix_produits,
        categorie: obj.categorie_produits,
        image: obj.image_produits,
        stock: obj.stock,
      }));
      const slice = posts.slice(
        this.state.offset,
        this.state.offset + this.state.perPage
      );
      const postData = slice.map((pd) => (
        <React.Fragment>
          <div className={`${pd.categorie}`}>
            <img
              className={"image"}
              src={`${pd.image}`}
              onClick={() => this.showModal(pd)}
              data-arg1={pd}
            />
            <div>{pd.nom}</div>
            <div>{pd.prix.toFixed(2)}€</div>
            <div className={`test${pd.stock}`}>stock : {pd.stock}</div>
          </div>
        </React.Fragment>
      ));
      this.setState({
        pageCount: Math.ceil(posts.length / this.state.perPage),

        postData,
      });
      this.setState({ posts });
      this.setState({ filter });
    });
    let i = 0;
    try {
      const posts = this.state.caisse.map((obj) => ({
        articleId: i++,
        id: obj.id,
        nom: obj.nom,
        prix: obj.prix,
        nombre: obj.nombre,
      }));
      prixTotal = 0;
      const articleShow = posts.map((pd) => (
        <React.Fragment>
          {this.calculerPrixTotal(pd.prix, pd.nombre)}
          <div>
            <div>
              {pd.nom} {pd.prix.toFixed(2)}€ x{pd.nombre} prix total :{" "}
              {(pd.prix * pd.nombre).toFixed(2)}€{" "}
            </div>
          </div>
          <hr></hr>
        </React.Fragment>
      ));
      this.setState({
        articleShow,
      });
    } catch {
      console.log("pas d'article");
    }
  }

  //fonction qui permet de changer de page
  handlePageClick = (e) => {
    const selectedPage = e.selected;
    const offset = selectedPage * this.state.perPage;

    this.setState(
      {
        currentPage: selectedPage,
        offset: offset,
      },
      () => {
        this.componentDidMount();
      }
    );
  };

  render() {
    return (
      <div>
        <div class="col-lg-3 col-sm-12 left">
          {this.state.articleShow}
          <div className="Total">Total à payer: {prixTotal.toFixed(2)} €</div>
          <Link to="/ticket">
            <button onClick={this.reset}>
              <span>ticket</span>
            </button>
          </Link>
        </div>
        <div class="col-lg-9 col-sm-12 right">
          <Modal
            show={this.state.show}
            handleClose={this.hideModal}
            validerModal={this.validerModal}
          >
            <div id="img1" />
            <div id="nom"></div>
            <div id="prix">€</div>
          </Modal>
          <Modal2 show2={this.state.show2} handleClose2={this.hideModal2}>
            <div id="img1" />
            <div id="nom"></div>
            <div id="prix">€</div>
          </Modal2>
          <div className="container">
            <div onChange={this.onChangeValue}>
              <input type="radio" value="Tout" name="categorie" /> Tout
              <input type="radio" value="fleur" name="categorie" /> fleur
              <br></br>
              <input type="radio" value="consommable" name="categorie" />
              Consommable <br></br>
              <input type="radio" value="décoration" name="categorie" />
              Décoration
            </div>

            {this.state.postData}
          </div>
          <ReactPaginate
            previousLabel={"prev"}
            nextLabel={"next"}
            breakLabel={"..."}
            breakClassName={"break-me"}
            pageCount={this.state.pageCount}
            marginPagesDisplayed={2}
            pageRangeDisplayed={5}
            onPageChange={this.handlePageClick}
            containerClassName={"pagination"}
            subContainerClassName={"pages pagination"}
            activeClassName={"active"}
          />
        </div>
      </div>
    );
  }
}
