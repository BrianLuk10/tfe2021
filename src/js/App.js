import { BrowserRouter, Route, Switch, Link } from "react-router-dom";
import React, { useState } from "react";
import Caisse from "../components/caisse/Caisse";
import Dashboard from "../components/dashbord/Dashboard";
import Connexion from "../components/connexion/Connexion";
import { Button } from "react-bootstrap";
import useToken from "../components/connexion/useToken";
import Ticket from "../components/caisse/Ticket";
import AjoutProduit from "../components/ajoutProduit/AjoutProduit";
import ModifierProduit from "../components/modifierProduit/modifierProduit";

export default function App() {
  /*
  const { token, setToken } = useToken();
  if (!token) {
    return <Connexion setToken={setToken} />
  }
  */
  return (
    <div className="wrapper">
      <h1>Application</h1>
      <BrowserRouter>
        <Link to="/caisse">
          <Button>
            <span>Caisse</span>
          </Button>
        </Link>
        <Link to="/dashboard">
          <Button>
            <span>Dashboard</span>
          </Button>
        </Link>
        <Link to="/modifierProduit">
          <Button>
            <span>Modifier un produit</span>
          </Button>
        </Link>
        <Link to="/ajoutProduit">
          <Button>
            <span>Ajouter un produit</span>
          </Button>
        </Link>
        <Switch>
          <Route path="/dashboard">
            <Dashboard />
          </Route>
          <Route path="/caisse">
            <Caisse />
          </Route>
          <Route path="/connexion">
            <Connexion />
          </Route>
          <Route path="/ticket">
            <Ticket />
          </Route>
          <Route path="/modifierProduit">
            <ModifierProduit />
          </Route>
          <Route path="/ajoutProduit">
            <AjoutProduit />
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
}
