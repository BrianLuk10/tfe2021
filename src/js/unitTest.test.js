import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import ReactDOM from "react-dom";
import * as React from "react";
import App from "./App";
import { MemoryRouter } from "react-router";
import Caisse from "../components/caisse/Caisse";
import Confirmer from "../components/modifierProduit/Confirmer";
import renderer from "react-test-renderer";
import ModifierProduit from "../components/modifierProduit/modifierProduit";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App></App>, div);
});

test("Page caisse", () => {
  render(
    <MemoryRouter>
      <Caisse></Caisse>
    </MemoryRouter>
  );
  const divText = screen.getByTestId("Test");
  expect(divText).not.toBeNull();
  expect(divText).toBeInTheDocument();
  expect(divText).toHaveTextContent("Total à payer: ");
});

test("message de confirmation", () => {
  const confirmer = () => {
    console.log("test");
  };
  const valeurTrue = true;
  render(<Confirmer show={valeurTrue} confirmer={confirmer}></Confirmer>);
  const divElement = screen.getByText("Action effectué");
  expect(divElement).not.toBeNull();
});

test(" match snapshot Confirmer", () => {
  const confirmer = () => {
    console.log("test");
  };
  const tree = renderer
    .create(<Confirmer show={true} confirmer={confirmer}></Confirmer>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("valeur dans recherche", () => {
  it("changer valeur de la barre de recherche", () => {
    const { queryByPlaceholderText } = render(
      <MemoryRouter>
        <ModifierProduit></ModifierProduit>
      </MemoryRouter>
    );
    const searchInput = queryByPlaceholderText("Rechercher...");
    fireEvent.change(searchInput, { target: { value: "test" } });
    expect(searchInput.value).toBe("test");
  });
});
