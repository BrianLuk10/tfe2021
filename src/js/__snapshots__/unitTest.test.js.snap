// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` match snapshot Confirmer 1`] = `
<div
  className="overlay"
>
  <div
    className="dialog"
  >
    <div
      className="dialog__content"
    >
      <h2
        className="dialog__title"
      >
        Action effectu√©
      </h2>
    </div>
    <hr />
    <div
      className="dialog__footer"
    >
      <button
        className="dialog__confirm"
        onClick={[Function]}
      >
        Confirmer
      </button>
    </div>
  </div>
</div>
`;
